File: src/solar/solar-wc3/util/game/JassBridgeUtil.ts
/** @noSelfInFile */
/**
 * 使用前置条件:
 * 需要在自定义函数加上这些jass声明
 *
 */

/** @noSelf */
export default class JassBridgeUtil {
    static sl_bridge: hashtable = require('jass.globals')["sl_bridge"]

    static getResultAsBoolean(): boolean {
        return LoadBoolean(JassBridgeUtil.sl_bridge, 2, 2);
    }

    static getResultAsInt(): number {
        return LoadInteger(JassBridgeUtil.sl_bridge, 2, 2);
    }

    static getResultAsReal(): number {
        return LoadReal(JassBridgeUtil.sl_bridge, 2, 2);
    }

    static getResultAsStr(): string {
        return LoadStr(JassBridgeUtil.sl_bridge, 2, 2);
    }


}

File: scripts/solar/solar-wc3/util/game/JassBridgeUtil.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 9,["7"] = 9,["8"] = 9,["10"] = 9,["11"] = 12,["12"] = 13,["13"] = 12,["14"] = 16,["15"] = 17,["16"] = 16,["17"] = 20,["18"] = 21,["19"] = 20,["20"] = 24,["21"] = 25,["22"] = 24,["23"] = 10});
local ____exports = {}
____exports.default = __TS__Class()
local JassBridgeUtil = ____exports.default
JassBridgeUtil.name = "JassBridgeUtil"
function JassBridgeUtil.prototype.____constructor(self)
end
function JassBridgeUtil.getResultAsBoolean()
    return LoadBoolean(____exports.default.sl_bridge, 2, 2)
end
function JassBridgeUtil.getResultAsInt()
    return LoadInteger(____exports.default.sl_bridge, 2, 2)
end
function JassBridgeUtil.getResultAsReal()
    return LoadReal(____exports.default.sl_bridge, 2, 2)
end
function JassBridgeUtil.getResultAsStr()
    return LoadStr(____exports.default.sl_bridge, 2, 2)
end
JassBridgeUtil.sl_bridge = require("jass.globals").sl_bridge
return ____exports
